### PR TITLE
Add colour support

### PR DESCRIPTION
**Pull request by rplociennik**
For what it's worth, this change adds colour support to the panel. Something that's been bugging me for a while now.

Seeing that this component has been acting as a demo component of sorts, I'm not sure how desirable this change might be...

### CODE REVIEW

File: foo_uie_console/main.cpp [hunk 1]
```diff
@@ -406,8 +430,20 @@ LRESULT ConsoleWindow::on_message(HWND wnd, UINT msg, WPARAM wp, LPARAM lp)
...
         break;
+    case WM_CTLCOLORSTATIC: {
+        HDC hdc = (HDC)wp;
+
+        cui::colours::helper helper(console_colour_id);
+
+        SetTextColor(hdc, helper.get_colour(cui::colours::colour_text));
+        SetBkColor(hdc, helper.get_colour(cui::colours::colour_background));
+
+        return (INT_PTR)s_brush;
```
Comment (reupen) [inline foo_uie_console/main.cpp:445]: I think the example from Microsoft Docs must be using `INT_PTR` for a dialog box procedure (`DLGPROC`), I'd switch to using a C++ cast to an `LRESULT`:

```suggestion
        return reinterpret_cast<LRESULT>(s_brush);
```

File: foo_uie_console/main.cpp [hunk 2]
```diff
@@ -406,8 +430,20 @@ LRESULT ConsoleWindow::on_message(HWND wnd, UINT msg, WPARAM wp, LPARAM lp)
         if (s_windows.empty()) {
             DeleteFont(s_font);
             s_font = nullptr;
+            DeleteObject(s_brush);
+            s_brush = nullptr;
         }
         break;
+    case WM_CTLCOLORSTATIC: {
```
Comment (reupen) [inline foo_uie_console/main.cpp:437]: Would be good to move the `WM_CTLCOLORSTATIC` case up to, say, just before `WM_ERASEBKGND` (seems like a more logical position based on the current order of the messages).

**Review by reupen — CHANGES_REQUESTED** [review]
This looks generally good to me 👍 

I've left some very minor comments. Let me know if you're OK to do those. (Also don't worry about the failing build – I'll fix that later.)

One thing I was a bit worried about was that the default background colour will change from grey to white (with default Windows colours), but I see even the Default UI does this now so I think it's fine.

**Comment by rplociennik** [comment]
Thanks for a swift review. I don't think there's a single remark I wouldn't agree with. I'll address these ASAP.

I bet the build process hasn't been updated since the component was last built. I figured it would be a non issue.

> One thing I was a bit worried about was that the default background colour will change from grey to white (with default Windows colours), but I see even the Default UI does this now so I think it's fine.

That also bothered me a little. I don't think that specifying default custom colours for a component would be that straightforward, but, then again, I'm not sure if any other component does that. All in all, I'm glad you won't miss the greys. ;)

**Review by reupen — APPROVED** [review]
Thanks 👍

**Pull request merged**